Integrate doctor

talatcanoguz opened the pull request.

Thomas-Chambers commented on lib/views/screens/addpatient_view.dart:
```diff
@@ -57,7 +60,13 @@ class AddPatientView extends HookWidget {
+                      ScaffoldMessenger.of(context).showSnackBar(
+                        SnackBar(content: Text(message)),
+                      );
+                    });
```
unrelated to this PR I suppose but the first/last name fields are not required to add patient. Not sure if intended or should fix. I am aware that we are not using it to validate the patient, not sure if we want to check that or if this is fine (this is probably fine for demo). Also for demo we can have two phones, one for patient and one for doctor. Just thought of that.

talatcanoguz replied:
No need to do that for demo.

Thomas-Chambers commented:
I am just doing normal testing. Is the forward arrow on recent activity supposed to pop up with the workout details like when you click on the calendar? Does not appear that there is any code for this but I felt as such

Thomas-Chambers commented:
Doctor view when I click into the text box to search it overflows and has this error in the bottom right 
![image](https://github.com/Senior-Design-Group-SP16-2023/mobile_app/assets/70411859/e3dba1a4-ef6a-4701-8df5-7cfb62305669)

Thomas-Chambers commented:
haha I break your code
![image](https://github.com/Senior-Design-Group-SP16-2023/mobile_app/assets/70411859/3936e4a9-e679-45b9-877b-a850f7e1f3e1)

Thomas-Chambers commented on lib/services/firestore_repository.dart:
```diff
@@ -79,4 +79,60 @@ class FireStoreRepository {
+    //Check if the given patient exists as a user
+    DocumentSnapshot patientSnapshot =
+        await _firestore.collection("users").doc(patientEmail).get();
+    if (patientSnapshot.exists) {
+      //Check if the given patient is a patient
+      bool? isPatient = patientSnapshot["isPatient"];
+      if (!isPatient!) {
+        message = "Given user is not a patient.";
+      } else {
+        //Check if the doctor has any patients
+        DocumentSnapshot doctorSnapshot =
```
We should check if the patient has completed signup before adding them. Otherwise we get null issues

talatcanoguz replied:
Good call, fixed it.

Thomas-Chambers commented:
The delete button on the doctor dashboard does not do anything

Thomas-Chambers commented:
Unsure why this is happening: 
![image](https://github.com/Senior-Design-Group-SP16-2023/mobile_app/assets/70411859/01871735-f9ae-4743-aa87-df7433cab21b)
Stack trace is 
`I/flutter ( 4471): []
I/flutter ( 4471): []

======== Exception caught by widgets library =======================================================
The following RangeError was thrown building RecentActivity(dirty):
RangeError (index): Invalid value: Valid value range is empty: 0

The relevant error-causing widget was: 
  RecentActivity RecentActivity:file:///C:/Users/thoma/eclipse-workspace/mobile_app/lib/views/screens/doctordashboarduser_view.dart:87:26
When the exception was thrown, this was the stack: 
#0      List.[] (dart:core-patch/growable_array.dart:264:36)
#1      RecentActivity.build (package:senior_design/views/widgets/recent_activity.dart:54:40)
#2      StatelessElement.build (package:flutter/src/widgets/framework.dart:5367:49)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5297:15)
#4      Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#5      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:5279:5)
#6      ComponentElement.mount (package:flutter/src/widgets/framework.dart:5273:5)
#7      Element.inflateWidget (package:flutter/src/widgets/framework.dart:4182:16)
#8      Element.updateChild (package:flutter/src/widgets/framework.dart:3701:20)
#9      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5322:16)
#10     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5462:11)
#11     Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#12     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2779:19)
#13     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:916:21)
#14     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:360:5)
#15     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1297:15)
#16     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1227:9)
#17     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1085:5)
#18     _invoke (dart:ui/hooks.dart:170:13)
#19     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:401:5)
#20     _drawFrame (dart:ui/hooks.dart:140:31)
====================================================================================================

======== Exception caught by widgets library =======================================================
The following StateError was thrown building RecentActivityGraphWidget(dirty, state: _RecentActivityGraphWidgetState#620b5):
Bad state: No element

The relevant error-causing widget was: 
  RecentActivityGraphWidget RecentActivityGraphWidget:file:///C:/Users/thoma/eclipse-workspace/mobile_app/lib/views/screens/doctordashboarduser_view.dart:113:26
When the exception was thrown, this was the stack: 
#0      ListBase.reduce (dart:collection/list.dart:187:22)
#1      _RecentActivityGraphWidgetState.build (package:senior_design/views/widgets/graph.dart:91:45)
#2      StatefulElement.build (package:flutter/src/widgets/framework.dart:5409:27)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5297:15)
#4      StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5462:11)
#5      Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#6      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:5279:5)
#7      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:5453:11)`

Thomas-Chambers commented:
I/flutter ( 4471): []
I/flutter ( 4471): []

======== Exception caught by widgets library =======================================================
The following RangeError was thrown building RecentActivity(dirty):
RangeError (index): Invalid value: Valid value range is empty: 0

The relevant error-causing widget was: 
  RecentActivity RecentActivity:file:///C:/Users/thoma/eclipse-workspace/mobile_app/lib/views/screens/doctordashboarduser_view.dart:87:26
When the exception was thrown, this was the stack: 
#0      List.[] (dart:core-patch/growable_array.dart:264:36)
#1      RecentActivity.build (package:senior_design/views/widgets/recent_activity.dart:54:40)
#2      StatelessElement.build (package:flutter/src/widgets/framework.dart:5367:49)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5297:15)
#4      Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#5      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:5279:5)
#6      ComponentElement.mount (package:flutter/src/widgets/framework.dart:5273:5)
#7      Element.inflateWidget (package:flutter/src/widgets/framework.dart:4182:16)
#8      Element.updateChild (package:flutter/src/widgets/framework.dart:3701:20)
#9      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5322:16)
#10     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5462:11)
#11     Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#12     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2779:19)
#13     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:916:21)
#14     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:360:5)
#15     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1297:15)
#16     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1227:9)
#17     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1085:5)
#18     _invoke (dart:ui/hooks.dart:170:13)
#19     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:401:5)
#20     _drawFrame (dart:ui/hooks.dart:140:31)
====================================================================================================

======== Exception caught by widgets library =======================================================
The following StateError was thrown building RecentActivityGraphWidget(dirty, state: _RecentActivityGraphWidgetState#620b5):
Bad state: No element

The relevant error-causing widget was: 
  RecentActivityGraphWidget RecentActivityGraphWidget:file:///C:/Users/thoma/eclipse-workspace/mobile_app/lib/views/screens/doctordashboarduser_view.dart:113:26
When the exception was thrown, this was the stack: 
#0      ListBase.reduce (dart:collection/list.dart:187:22)
#1      _RecentActivityGraphWidgetState.build (package:senior_design/views/widgets/graph.dart:91:45)
#2      StatefulElement.build (package:flutter/src/widgets/framework.dart:5409:27)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5297:15)
#4      StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5462:11)
#5      Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#6      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:5279:5)
#7      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:5453:11)

Thomas-Chambers commented:
> ======== Exception caught by widgets library =======================================================
The following StateError was thrown building RecentActivityGraphWidget(dirty, state: _RecentActivityGraphWidgetState#b74a5):
Bad state: No element

The relevant error-causing widget was: 
  RecentActivityGraphWidget RecentActivityGraphWidget:file:///C:/Users/thoma/eclipse-workspace/mobile_app/lib/views/screens/doctordashboarduser_view.dart:113:26
When the exception was thrown, this was the stack: 
#0      ListBase.reduce (dart:collection/list.dart:187:22)
#1      _RecentActivityGraphWidgetState.build (package:senior_design/views/widgets/graph.dart:91:45)
#2      StatefulElement.build (package:flutter/src/widgets/framework.dart:5409:27)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5297:15)
#4      StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5462:11)
#5      Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#6      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:5279:5)
#7      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:5453:11)
#8      ComponentElement.mount (package:flutter/src/widgets/framework.dart:5273:5)
#9      Element.inflateWidget (package:flutter/src/widgets/framework.dart:4182:16)
#10     Element.updateChild (package:flutter/src/widgets/framework.dart:3701:20)
#11     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5322:16)
#12     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5462:11)
#13     Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#14     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2779:19)
#15     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:916:21)
#16     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:360:5)
#17     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1297:15)
#18     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1227:9)
#19     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1085:5)
#20     _invoke (dart:ui/hooks.dart:170:13)
#21     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:401:5)
#22     _drawFrame (dart:ui/hooks.dart:140:31)
====================================================================================================

======== Exception caught by widgets library =======================================================
The following RangeError was thrown building RecentActivity(dirty):
RangeError (index): Invalid value: Valid value range is empty: 0

The relevant error-causing widget was: 
  RecentActivity RecentActivity:file:///C:/Users/thoma/eclipse-workspace/mobile_app/lib/views/screens/doctordashboarduser_view.dart:87:26
When the exception was thrown, this was the stack: 
#0      List.[] (dart:core-patch/growable_array.dart:264:36)
#1      RecentActivity.build (package:senior_design/views/widgets/recent_activity.dart:54:40)
#2      StatelessElement.build (package:flutter/src/widgets/framework.dart:5367:49)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5297:15)
#4      Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#5      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:5279:5)
#6      ComponentElement.mount (package:flutter/src/widgets/framework.dart:5273:5)
#7      Element.inflateWidget (package:flutter/src/widgets/framework.dart:4182:16)
#8      Element.updateChild (package:flutter/src/widgets/framework.dart:3701:20)
#9      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5322:16)
#10     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5462:11)
#11     Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#12     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2779:19)
#13     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:916:21)
#14     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:360:5)
#15     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1297:15)
#16     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1227:9)
#17     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1085:5)
#18     _invoke (dart:ui/hooks.dart:170:13)
#19     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:401:5)
#20     _drawFrame (dart:ui/hooks.dart:140:31)

talatcanoguz commented:
1. No need to create a new pop-up, all the info is on the recent activity card. I removed the button.
2. Not sure why this happens, told @nealhmakwana about it.
3. I think it is best to leave this for later because it is not needed for demo.
4. Good call, fixed this.
5. This is fixed in main.

Lmk what you think @Thomas-Chambers

Thomas-Chambers reviewed (approved):
lgtm!
